src/main/java/io/appform/dropwizard/sharding/dao/LookupDao.java

```diff
@@ -385,6 +417,46 @@ public List<T> scatterGather(DetachedCriteria criteria) {
                 }).flatMap(Collection::stream).collect(Collectors.toList());
     }
 
+    /**
+     * Queries using the specified criteria across all shards and returns the result.
+     * <b>Note:</b> This method runs the query serially and it's usage is not recommended.
+     *
+     * @param querySpec The select criteria
+     * @return List of elements or empty if none match
+     */
+    public List<T> scatterGather(final QuerySpec<T, T> querySpec) {
+        return IntStream.range(0, daos.size())
+                .mapToObj(shardId -> {
+                    try {
+                        val dao = daos.get(shardId);
+                        return transactionExecutor.execute(dao.sessionFactory, true, dao::select, querySpec, "scatterGather",
+                                shardId);
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                }).flatMap(Collection::stream).collect(Collectors.toList());
+    }
+
+    /**
+     * Queries using the specified criteria across all shards and returns the result.
+     * <b>Note:</b> This method runs the query serially and it's usage is not recommended.
+     *
+     * @param querySpec The select criteria
+     * @return List of elements or empty if none match
+     */
+    public List<T> scatterGather(final QuerySpec<T, T> querySpec, int start, int numRows) {
```

Comment:
Why two functions? Pagination does not work here actually .. this will give unpredictable results. It's better to not create this and move people to the `scrollUp` and `scrollDown` apis

Reply:
Removed this